### PR TITLE
python-for-android rebase

### PR DESCRIPTION
**Pull request by dbnicholson**
This updates the build to use the recent p4a 2022.09.04 release with our downstream changes. Mostly this is uneventful but a couple workarounds can be dropped since they've been fixed in p4a.

https://phabricator.endlessm.com/T33884

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -165,7 +165,7 @@ kolibri.apk: p4a_android_distro src/kolibri src/apps-bundle src/collections asse
 	@echo "--- :android: Build APK"
 	$(P4A) apk --release --sign $(ARCH_OPTIONS) --version=$(APK_VERSION) --numeric-version=$(BUILD_NUMBER)
 	mkdir -p dist
-	mv kolibri-release-$(APK_VERSION)-.apk dist/kolibri-release-$(APK_VERSION).apk
+	mv kolibri-release-$(APK_VERSION).apk dist/kolibri-release-$(APK_VERSION).apk
```
Comment (manuq) [inline Makefile:164]: :+1:

File: p4a-recipes/zstandard/__init__.py [hunk 2]
```diff
@@ -14,14 +12,7 @@ class ZStandardRecipe(CompiledComponentsPythonRecipe):
...
     depends = ["cffi", "setuptools"]
-
-    # Local patches need to be an absolute paths or they won't be found
-    # by patch.
-    #
-    # https://github.com/kivy/python-for-android/issues/2623
-    patches = [
-        os.path.realpath(os.path.join(os.path.dirname(__file__), "preprocessor.patch")),
-    ]
+    patches = ["preprocessor.patch"]
```
Comment (manuq) [inline p4a-recipes/zstandard/__init__.py:15]: :+1:

File: src/main.py [hunk 3]
```diff
@@ -3,6 +3,7 @@
 import time
 
 import initialization  # noqa: F401 keep this first, to ensure we're set up for other imports
+from android.activity import register_activity_lifecycle_callbacks
```
Comment (manuq) [inline src/main.py:6]: :+1:

**Review by manuq — APPROVED** [review]
Looking good. Jenkins failed with:
> ERROR: Package 'python-for-android' requires a different Python: 3.6.9 not in '>=3.7.0'

**Comment by dbnicholson** [comment]
> Looking good. Jenkins failed with:
> 
> > ERROR: Package 'python-for-android' requires a different Python: 3.6.9 not in '>=3.7.0'

Oh. I didn't notice that locally. I guess this is because the docker image is using Ubuntu bionic and that has 3.6.3. I guess I only tested in the toolbox container.

**Comment by dbnicholson** [comment]
> Oh. I didn't notice that locally. I guess this is because the docker image is using Ubuntu bionic and that has 3.6.3. I guess I only tested in the toolbox container.

I rewrote the non-toolbox Dockerfile so it was almost the same as the toolbox image. It worked in a local build and I think it should work in Jenkins. I like that better since the build environment in Jenkins is then closer to what we use locally.

File: Makefile [hunk 4]
```diff
@@ -61,16 +61,12 @@ clean-whl:
 
 .PHONY: get-whl
 get-whl: clean-whl
-# The eval and shell commands here are evaluated when the recipe is parsed, so we put the cleanup
-# into a prerequisite make step, in order to ensure they happen prior to the download.
-	$(eval DLFILE = $(shell wget --content-disposition -P whl/ "${whl}" 2>&1 | grep "Saving to: " | sed 's/Saving to: ‘//' | sed 's/’//'))
-	$(eval WHLFILE = $(shell echo "${DLFILE}" | sed "s/\?.*//"))
-	[ "${DLFILE}" = "${WHLFILE}" ] || mv "${DLFILE}" "${WHLFILE}"
+	wget -O whl/kolibri.whl "${whl}"
```
Comment (manuq) [inline Makefile:64]: Yes this simplify things. :+1:

**Review by manuq — COMMENTED** [review]
I really like that the Dockerfile is closer to the toolbox now.

**Pull request merged**